src/microbe_stage/organelle_unlocks/StatisticBasedUnlockCondition.cs

```diff
@@ -0,0 +1,188 @@
+﻿namespace UnlockConstraints
+{
+    using Newtonsoft.Json;
+
+    /// <summary>
+    ///   An unlock condition that relies on a certain <see cref="IStatistic"/> to track progress
+    /// </summary>
+    public abstract class StatisticBasedUnlockCondition : IUnlockCondition
+    {
+        [JsonIgnore]
+        public abstract StatsTrackerEvent RelevantEvent { get; }
+
+        [JsonIgnore]
+        public IStatistic? RelevantStatistic { get; set; }
+
+        public abstract bool Satisfied();
+
+        public abstract void GenerateTooltip(LocalizedStringBuilder builder);
+
+        public virtual void Check(string name)
+        {
+        }
+
+        public virtual void Resolve(SimulationParameters parameters)
+        {
+        }
+    }
+
+    /// <summary>
+    ///   The number of microbes engulfed by the player is above the specified amount.
+    /// </summary>
+    public class EngulfedMicrobesAbove : StatisticBasedUnlockCondition
+    {
+        public override StatsTrackerEvent RelevantEvent => StatsTrackerEvent.PlayerEngulfedOther;
```

Comment:
All of these properties that have a fixed value need to have json ignore on them, otherwise these are unnecessarily written out into the save data.

Reply:
From what I can tell, this is never actually loaded into saves, just loaded from the registry.

But for futureproofing and clarity I'll add the attribute anyway

Reply:
So there's the statistics data and this unlock condition as separate classes? And one of those is saved while this is not. I guess that is probably good design as many unlocks do need similar kind of information so adding that one layer of decoupling might be necessary.